blocks/leaphy/leaphy_extra.js

```diff
@@ -33,358 +33,351 @@ Blockly.Constants.Colour.HUE = 20;
-    this.appendValueInput("LED_BASIC_GREEN", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_BASIC_GREEN)
-      .setCheck('Number');
-    this.appendValueInput("LED_BASIC_BLUE", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_BASIC_BLUE)
-      .setCheck('Number');
-    this.setInputsInline(true);
-    this.setPreviousStatement(true, null);
-    this.setNextStatement(true, null);
-    this.setStyle('leaphy_blocks');
-  }
-};
-
-Blockly.Blocks["leaphy_led_set_speed"] = {
-  init: function () {
-    this.appendDummyInput()
-      .appendField(Blockly.Msg.LEAPHY_LED_SET_SPEED);
-    this.appendValueInput("LED_SET_SPEEDVALUE", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_SET_SPEEDVALUE)
-      .setCheck('Number');
-    this.setInputsInline(true);
-    this.setPreviousStatement(true, null);
-    this.setNextStatement(true, null);
-    this.setStyle('leaphy_blocks');
-  }
-};
-
-Blockly.Blocks["leaphy_led_strip_demo"] = {
-  init: function () {
-    this.appendDummyInput()
-      .appendField(Blockly.Msg.LEAPHY_LED_STRIP_DEMO)
-      .appendField(new Blockly.FieldDropdown(ledstripDemoOptions), "DEMO_TYPE");
-    this.appendValueInput("LED_STRIP_DEMO_RED", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_STRIP_DEMO_RED)
-      .setCheck('Number');
-    this.appendValueInput("LED_STRIP_DEMO_GREEN", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_STRIP_DEMO_GREEN)
-      .setCheck('Number');
-    this.appendValueInput("LED_STRIP_DEMO_BLUE", 'Number')
-      .appendField(Blockly.Msg.LEAPHY_LED_STRIP_DEMO_BLUE)
-      .setCheck('Number');
-    this.setInputsInline(true);
-    this.setPreviousStatement(true, null);
-    this.setNextStatement(true, null);
-    this.setStyle('leaphy_blocks');
-  }
-};
-
-Blockly.Blocks['leaphy_servo_write'] = {
-  /**
-     * Block for writing an angle value into a servo pin.
-     * @this Blockly.Block
-     */
-  init: function () {
-    this.setHelpUrl('http://arduino.cc/en/Reference/ServoWrite');
-    this.setColour(Blockly.Blocks.servo.HUE);
-    this.appendDummyInput()
-      .appendField(Blockly.Msg.ARD_SERVO_WRITE)
-      .appendField(new Blockly.FieldDropdown(digitalPinOptions), 'SERVO_PIN');
-    this.setInputsInline(false);
-    this.appendValueInput('SERVO_ANGLE')
-      .setCheck(Blockly.Types.NUMBER.checkList)
-      .appendField(Blockly.Msg.ARD_SERVO_WRITE_TO);
-    this.appendDummyInput()
-      .appendField(Blockly.Msg.ARD_SERVO_WRITE_DEG_180);
-    this.setInputsInline(true);
-    this.setPreviousStatement(true, null);
-    this.setNextStatement(true, null);
-    this.setTooltip(Blockly.Msg.ARD_SERVO_WRITE_TIP);
-    this.setStyle('leaphy_blocks');
+Blockly.defineBlocksWithJsonArray([
+  {
+    "type": "leaphy_rgb_color",
+    "message0": "%1",
+    "args0": [
+      {
+        "type": "field_dropdown",
+        "name": "COLOR_TYPE ",
+        "options": [["%{BKY_LEAPHY_RGB_COLOR_RED}", "0"], ["%{BKY_LEAPHY_RGB_COLOR_GREEN}", "1"], ["%{BKY_LEAPHY_RGB_COLOR_BLUE}", "2"]]
```

Comment:
It should be possible to reference a variable here, right? So you can prevent repeating yourself with these arrays.